### PR TITLE
cargo-edit: fix Darwin build

### PR DESCRIPTION
**Pull request by happysalada**
##### Motivation for this change

Latest version was not building on darwin

###### Things done

Originally my intent was only to change things on darwin platform.
It appears though that libz was missing as a dependency.
Also in a lot of recent rust builds, cc crate attempts to run xcbuild.
I also run nixfmt, so some format has changed a little bit.

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [X] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/package-management/cargo-edit/default.nix
Comment (danieldk) [inline pkgs/tools/package-management/cargo-edit/default.nix:19]: This should not be necessary once staging is merged in master:

https://github.com/NixOS/nixpkgs/pull/97000

Maybe add a reference + comment that this should be removed once that PR lands?

Also, the `xcbuild` dependency should (probably) be conditional so that's only used on Darwin.

File: pkgs/tools/package-management/cargo-edit/default.nix
Comment (danieldk) [inline pkgs/tools/package-management/cargo-edit/default.nix:35]: Are these changes necessary? Did nixpkgs-fmt make them?

Comment (happysalada) [reply pkgs/tools/package-management/cargo-edit/default.nix:35]: yes, I just ran the formatter.

File: pkgs/tools/package-management/cargo-edit/default.nix
Comment (happysalada) [reply pkgs/tools/package-management/cargo-edit/default.nix]: pushed a fix, let me know if that works for you.

File: pkgs/tools/package-management/cargo-edit/default.nix
Comment (danieldk) [reply pkgs/tools/package-management/cargo-edit/default.nix]: Strange, I don't get that formatting with `nixpkgs-fmt`.

File: pkgs/tools/package-management/cargo-edit/default.nix
Comment (danieldk) [reply pkgs/tools/package-management/cargo-edit/default.nix]: Thanks! I have pushed a change that makes the formatting a little more canonical. While at it, I tacked on another commit that passes the `Security` framework in the preferred manner.

**Review by danieldk — APPROVED** [review]
Should be good to go once ofborg is also happy.

File: pkgs/tools/package-management/cargo-edit/default.nix
Comment (happysalada) [reply pkgs/tools/package-management/cargo-edit/default.nix]: Thanks!
I didn't know that was the prefered manner. Will keep it in mind for next time.

**Pull request merged**
